app/com/linkedin/drelephant/clients/azkaban/AzkabanJobStatusUtil.java

```diff
@@ -70,7 +96,8 @@ public AzkabanJobStatusUtil(String url) {
    * @throws URISyntaxException
    */
   public Map<String, String> getJobsFromFlow(String execUrl) throws MalformedURLException, URISyntaxException {
-    _workflowClient.setURL(execUrl);
-    return _workflowClient.getJobsFromFlow();
+    AzkabanWorkflowClient workflowClient = getWorkflowClient(execUrl);
+    workflowClient.setURL(execUrl);
```

Comment:
Is this required? Doesn't the previous line already return the updated client for the url.

Reply:
No, first line is only returning the workflowClient, next line is setting the url.

Reply:
The url seems to be already set when you call `getWorkflowClient(execUrl)`. The explicit setURL in the next line seems redundant.

Reply:
If you are talking about this code getWorkflowClientInstance(scheduler, url), then this is only getting called when first time WorkflowClient is created, after that it will always pick up from the map.